src/models/room.ts

```diff
@@ -904,6 +929,71 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
         });
     }
 
+    /**
+     * Empty out the current live timeline and re-request it. This is used when
+     * historical messages are imported into the room via MSC2716 `/batch_send
+     * because the client may already have that section of the timeline loaded.
+     * We need to force the client to throw away their current timeline so that
+     * when they back paginate over the area again with the historical messages
+     * in between, it grabs the newly imported messages. We can listen for
+     * `UNSTABLE_MSC2716_MARKER`, in order to tell when historical messages are ready
+     * to be discovered in the room and the timeline needs a refresh. The SDK
+     * emits a `RoomEvent.historyImportedWithinTimeline` event when we detect a
+     * valid marker and can check the needs refresh status via
+     * `room.getTimelineNeedsRefresh()`.
+     */
+    public async refreshLiveTimeline(): Promise<void> {
+        const liveTimelineBefore = this.getLiveTimeline();
+        const forwardPaginationToken = liveTimelineBefore.getPaginationToken(EventTimeline.FORWARDS);
+        const eventsBefore = liveTimelineBefore.getEvents();
+        const mostRecentEventInTimeline = eventsBefore[eventsBefore.length - 1];
+        logger.log(
+            `[refreshLiveTimeline] for room ${this.roomId} at ` +
+            `mostRecentEventInTimeline=${mostRecentEventInTimeline.getId()} ` +
+            `forwardPaginationToken+${forwardPaginationToken}`,
+        );
+
+        // Empty out all of `this.timelineSets` but still keeps the same
+        // `timelineSet` references around so the React code updates properly
+        // and doesn't ignore the room events we emit because it checks that the
+        // `timelineSet` references are the same. We need the `timelineSet`
+        // empty so that the `client.getEventTimeline(...)` call later, will
+        // call `/context` and create a new timeline instead of returning the
+        // same one.
+        this.resetLiveTimeline(null, null);
+
+        // Get a reference to the emptied out `timelineSet`
+        //
+        // TODO: Do we want to use `this.getLiveTimeline().getTimelineSet()`
```

Comment:
Needs resolving

Reply:
I'm looking for advice here. What are your thoughts?

Reply:
`this.getLiveTimeline().getTimelineSet()` is identical to `this.getUnfilteredTimelineSet()` as the former is defined as `this.getUnfilteredTimelineSet().getLiveTimeline()` and an event-timeline is only part of one event-timeline-set 

So might as well go with the current approach which involves less indirection